crates/rpc-types/src/genesis.rs

```diff
@@ -1,5 +1,42 @@
 //! OP types for genesis data.
 
+use alloy_serde::OtherFields;
+use serde::de::Error;
+
+/// Info for Optimism chain.
+#[derive(Default, Debug, Clone, Copy, Eq, PartialEq, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct OptimismChainInfo {
+    /// Genesis info
+    pub genesis_info: Option<OptimismGenesisInfo>,
+    /// Base fee info
+    pub base_fee_info: Option<OptimismBaseFeeInfo>,
+}
+
+impl OptimismChainInfo {
+    /// Extract the Optimism chain info from `OtherFields` object.
```

Comment:
needs additional context that this is expected to be other, optimism specific fields of `genesis.config`